tests/integration/test_metric_collection.py

```diff
@@ -0,0 +1,101 @@
+    start_time = time.time()
+
+    with open('expected_output.json', "r", newline='\n') as file_with_expected:
+        expected_json_raw = json.load(file_with_expected)
+
+    metric_matches = False
+    length_matches = False
+    while time.time() - start_time < timeout:
+        response = None
+        try: 
+            response = requests.get(url)
+            response.raise_for_status()
+        except requests.exceptions.RequestException as e:
+            print(f"An error occurred while making the request: {e}")
+
+        if response is not None and response.status_code == 200:
+            processed_successfully = True
+            try:
+                print("Successfully downloaded!")
+                lines = response.content.splitlines()
+                metrics = [json.loads(line) for line in lines]
+                merged_json = merge_jsons(metrics)
+
+                actual_json = json.dumps(merged_json, sort_keys=True, indent=2)
+                expected_json = json.dumps(expected_json_raw, sort_keys=True, indent=2)
+
+                write_actual = os.getenv("WRITE_ACTUAL_JSON", "False")
+                if write_actual == "True":
+                    with open("actual.json", "w", newline='\n') as f:
+                        f.write(actual_json)
+
+                length_matches = False
+                if len(actual_json.splitlines()) == len(expected_json.splitlines()):
+                    print(
+                        f'LineCount of outputs matches, expected: {len(expected_json.splitlines())}, actual: {len(actual_json.splitlines())}')
+                    length_matches = True
+                else:
+                    print(
+                        f'LineCount of outputs does not match, expected: {len(expected_json.splitlines())}, actual: {len(actual_json.splitlines())}')
+
+                metric_names = get_unique_metric_names(merged_json)
+                metric_matches = False
+                if all(name in metric_names for name in expected_metric_names):
+                    print("All specific metric names are found in the response.")
+                    metric_matches = True
+                else:
+                    print('Some specific metric names are not found in the response')
+                    print(f'Expected: {expected_metric_names}')
```

Comment:
Wouldn't this spam logs too much? Maybe just past what does miss.

Reply:
Good point, adjusted